cpp/src/io/parquet/reader_impl.cu

```diff
@@ -657,19 +699,13 @@ table_with_metadata reader::impl::read(int skip_rows, int num_rows, int row_grou
             column_types[i].id(), _timestamp_type.id(), col_schema.type,
             col_schema.converted_type, col_schema.type_length);
 
-        uint8_t *d_compdata = nullptr;
-        if (col_meta.total_compressed_size != 0) {
-          const auto offset = (col_meta.dictionary_page_offset != 0)
+        column_chunk_offsets[chunks.size()] = (col_meta.dictionary_page_offset != 0)
```

Comment:
No: the offset vector is pre-allocated at the max size, whereas the current value of chunk.size() is the position of the new chunk about to be added (the subsequent `hostdevice_vector` 's `insert()` call is really a sort of `push_back`)

Reply:
Ah, I see. Nevermind then!
I like that this `hostdevice_vector` constructor has a parameter for the reserved size. It would be convenient if std and thrust vectors had that as well...